### PR TITLE
Configurable the daemon resource requirements

### PR DESCRIPTION
**Pull request by ccojocar**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

It makes configurable the daemon resource requirements. This is useful when the daemon requires different resource requirements due to the cluster size (e.g. number of pods is watching for).

#### Does this PR introduce a user-facing change?

```release-note
Make the daemon resource requirements configurable.

```

### CODE REVIEW

**Comment by jhrozek** [comment]
It would be nice to document the new options. I'm on the fence about a test, ideally we should have everything tested (and e.g. we do have tests for webhook label changes), but the patch "just" adds a public kube type, so..*shrug*.

By the way, thanks a lot for doing this performance and scalability work!

File: internal/pkg/manager/spod/spod_controller.go [hunk 1]
```diff
@@ -472,6 +472,11 @@ func (r *ReconcileSPOd) getConfiguredSPOd(
 		}
 	}
 
+	// Overwrite the SPOD's default resoruce requirements
```
Comment (pjbgf) [inline internal/pkg/manager/spod/spod_controller.go:475]:
```suggestion
	// Overwrite the SPOD's default resource requirements
```

File: internal/pkg/manager/spod/spod_controller.go [hunk 2]
```diff
@@ -472,6 +472,11 @@ func (r *ReconcileSPOd) getConfiguredSPOd(
...
+	if cfg.Spec.DaemonResourceRequirements != nil {
+		templateSpec.Containers[bindata.ContainerIDDaemon].Resources = *cfg.Spec.DaemonResourceRequirements
+	}
```
Comment (pjbgf) [inline internal/pkg/manager/spod/spod_controller.go:478]: I think we could extend this to the `initContainers` as well. wdyt?

File: api/spod/v1alpha1/spod_types.go [hunk 3]
```diff
@@ -106,6 +106,11 @@ type SPODSpec struct {
 	// namespace to use for pulling the images from SPOD pod from a private registry.
 	// +optional
 	ImagePullSecrets []corev1.LocalObjectReference `json:"imagePullSecrets,omitempty"`
+
+	// DaemonResourceRequirements if defined, overwrites the default resoruce requirements
```
Comment (pjbgf) [inline api/spod/v1alpha1/spod_types.go:110]:
```suggestion
	// DaemonResourceRequirements if defined, overwrites the default resource requirements.
```

**Review by pjbgf — COMMENTED** [review]
A few small nits, but overall lgtm from me. Thank you for working on this @ccojocar.

File: internal/pkg/manager/spod/spod_controller.go (hunk 1 above)
Comment (ccojocar) [reply internal/pkg/manager/spod/spod_controller.go]: fixed

File: internal/pkg/manager/spod/spod_controller.go (hunk 2 above)
Comment (ccojocar) [reply internal/pkg/manager/spod/spod_controller.go:478]: Do you mean the same resources?

**Comment by ccojocar** [comment]
> It would be nice to document the new options. I'm on the fence about a test, ideally we should have everything tested (and e.g. we do have tests for webhook label changes), but the patch "just" adds a public kube type, so..shrug.


I'll add some docs and tests.

File: internal/pkg/manager/spod/spod_controller.go (hunk 2 above)
Comment (ccojocar) [reply internal/pkg/manager/spod/spod_controller.go:478]: I think the two init containers don't have really performance issues, or need more resources. I would let them as they are for now.

**Comment by ccojocar** [comment]
@jhrozek @pjbgf I think that I addressed all your remarks. I added some documentation and an e2e test.

File: api/spod/v1alpha1/spod_types.go [hunk 4]
```diff
@@ -308,6 +308,11 @@ type SPODSpec struct {
 	// namespace to use for pulling the images from SPOD pod from a private registry.
 	// +optional
 	ImagePullSecrets []corev1.LocalObjectReference `json:"imagePullSecrets,omitempty"`
+
+	// DaemonResourceRequirements if defined, overwrites the default resource requirements
+	// of SPOD daemon.
+	// +optinal
```
Comment (JAORMX) [inline api/spod/v1alpha1/spod_types.go:314]: `s/optinal/optional/`

**Review by JAORMX — CHANGES_REQUESTED** [review]

File: internal/pkg/manager/spod/spod_controller.go (hunk 2 above)
Comment (pjbgf) [reply internal/pkg/manager/spod/spod_controller.go:478]: The idea was mostly to provide users a way to set resource requests/limits across all containers we expose. But I agree that our current init containers are less resource intensive, so happy to not expand this for the time being. :+1:

File: installation-usage.md [hunk 5]
```diff
@@ -231,6 +232,16 @@ The `ds/spod` should now be updated by the manager with the new SELinux type, an
             type: unconfined_t
 ```
 
+## Customise the daemon resource requirements
+
+The default resource requirements of the daemon container can be adjusted in the SPOD configuration in the
+`daemonResourceRequirements` filed as follows:
```
Comment (pjbgf) [inline installation-usage.md:238]: nit:
```suggestion
The default resource requirements of the daemon container can be adjusted by using the field `daemonResourceRequirements`
from the SPOD configuration as follows:
```

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**